service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -266,7 +281,8 @@ AccessURL fetchDrsObjectAccessUrl(
       List<DrsHubAuthorization> drsHubAuthorizations,
       AuditLogEvent.Builder auditLogEventBuilder,
       String ip,
-      String googleProject) {
+      String googleProject,
+      String transactionId) {
 
     var drsApi = drsApiFactory.getApiFromUriComponents(uriComponents, drsProvider);
```

Comment:
You could probably support setting the header conditionally if its TDR from within `getApiFromUriComponents`